src/analysis/types.js

```diff
@@ -2,17 +2,70 @@
 
 import {type NodeAddressT, type EdgeAddressT} from "../core/graph";
 
+/**
+ * Represents a "Type" of node in the graph.
+ *
+ * An node is considered a member of this type if
+ * its address has this type's prefix as an address prefix,
+ * according to the semantics of `Address.hasPrefix`.
+ *
+ * When an edge matches multiple types, we will usually only
+ * consider the most specific matching type (i.e. the match
+ * with the longest prefix).
+ *
+ * Node types are used for organization in the UI, and for
+ * applying type-level weights.
+ */
+export type NodeType = {|
+  // The name for an individual node of this type.
+  // For example, for the GitHub PULL_REQUEST node type, the name is "Pull
+  // request". The first letter of the name should be capitalized.
+  +name: string,
+  // The name for multiple nodes of this type.
+  // For example, for the GitHub PULL_REQUEST node type, the pluralName is
+  // "Pull requests". The first letter of this name should be capitalized.
+  +pluralName: string,
+  // The address prefix that will be used to test whether a node is a member
```

Comment:
Might be nice to specify here and bring it back around for your reader: "`NodeAddressT` is the address prefix that will be used to test whether a node is a member of this NodeType".

Reply:
Rewrote it to be more explicit.